### PR TITLE
matrix-appservice-irc 0.26.1 -> 0.30.0

### PR DESCRIPTION
**Pull request by DavHau**
###### Motivation for this change
Hackint matrix irc bridge broke because this package is outdated

###### Things done
- switch to using upstream package-lock.json (build errored out without it)
- for node2nix instead of referencing a remote project via package.json, the package.json is now downloaded (because https://github.com/svanderburg/node2nix/issues/207)
- the `update-sh` script is updated accordingly

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by andir** [comment]
@ofborg test matrix-appservice-irc

**Comment by DavHau** [comment]
EditorConfig check fails because the REVISION file doesn't end with a newline. But this is done on purpose since the version should not contain a new line. Is there a way to make an exception for the test?

File: pkgs/servers/matrix-synapse/matrix-appservice-irc/default.nix [hunk 1]
```diff
@@ -1,16 +1,23 @@
-{ pkgs, nodePackages, makeWrapper, nixosTests, nodejs, stdenv, lib, ... }:
+{ pkgs, nodePackages, makeWrapper, nixosTests, nodejs, stdenv, lib, fetchFromGitHub, ... }:
```
Comment (andir) [inline pkgs/servers/matrix-synapse/matrix-appservice-irc/default.nix:1]: We should drop the `...` here. It isn't required and not good practice. I know this isn't your fault but if we change it we can make a slight improvement.

**Comment by mweinelt** [comment]
```patch
diff --git a/nixos/tests/matrix-appservice-irc.nix b/nixos/tests/matrix-appservice-irc.nix
index 79b07ef83c5..264142a9066 100644
--- a/nixos/tests/matrix-appservice-irc.nix
+++ b/nixos/tests/matrix-appservice-irc.nix
@@ -131,6 +131,8 @@ import ./make-test-python.nix ({ pkgs, ... }:
     };
 
     testScript = ''
+      import pathlib
+
       start_all()
 
       ircd.wait_for_unit("ngircd.service")
```

File: pkgs/servers/matrix-synapse/matrix-appservice-irc/update.sh [hunk 2]
```diff
@@ -15,10 +15,16 @@ fi
 
 echo "matrix-appservice-irc: $CURRENT_VERSION -> $TARGET_VERSION"
 
-sed -i "s/#$CURRENT_VERSION/#$TARGET_VERSION/" package.json
+# sed -i "s/#$CURRENT_VERSION/#$TARGET_VERSION/" package.json
+rm -f package.json package-lock.json
+wget https://github.com/matrix-org/matrix-appservice-irc/raw/$TARGET_VERSION/package.json
+wget -O package-lock-temp.json https://github.com/matrix-org/matrix-appservice-irc/raw/$TARGET_VERSION/package-lock.json
+echo -n "$TARGET_VERSION" > ./REVISION
```
Comment (andir) [inline pkgs/servers/matrix-synapse/matrix-appservice-irc/update.sh:22]: Just give it a newline and trim / get the first line in the nix revision. That makes this part more robust to manual updates & gets rid of the editorconfig error.

**Comment by DavHau** [comment]
> ```diff
> nixos/tests/matrix-appservice-irc.nix
> ```

Thanks. I overlooked the test.
After fixing the import, still erroring out with:

```
client # Traceback (most recent call last):
client #   File "/run/current-system/sw/bin/do_test", line 7, in <module>
client #     matrix.register_with_password(username="alice", password="foobar")
client #   File "/nix/store/630jdrv3dxh67gvzd3ca9xmc2cn7an4w-python3-3.9.6-env/lib/python3.9/site-packages/matrix_client/client.py", line 203, in register_with_password
client #     response = self.api.register(
client #   File "/nix/store/630jdrv3dxh67gvzd3ca9xmc2cn7an4w-python3-3.9.6-env/lib/python3.9/site-packages/matrix_client/api.py", line 160, in register
client #     return self._send(
client #   File "/nix/store/630jdrv3dxh67gvzd3ca9xmc2cn7an4w-python3-3.9.6-env/lib/python3.9/site-packages/matrix_client/api.py", line 748, in _send
client #     raise MatrixRequestError(
client # matrix_client.errors.MatrixRequestError: 400: {"errcode":"M_UNKNOWN","error":"An access token should not be provided on requests to /register (except if type is m.login.application_service)"}
```

For me it looks like a bug in the matrix_client python package.
Our code is very simple and we do definitely not pass any access token:
```python
import socket
from matrix_client.client import MatrixClient
from time import sleep

matrix = MatrixClient("${homeserverUrl}")
matrix.register_with_password(username="alice", password="foobar")  // <-- it fails here
...
```

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 137035` at db2264b4 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>matrix-appservice-irc</li>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/development/node-packages/node-env.nix:410:14:
```
    |
410 |       inherit dontNpmInstall preRebuild unpackPhase buildPhase;
    |              ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

util-linux is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/node-packages/node-env.nix:402:7:
```
    |
402 |       buildInputs = [ tarWrapper python nodejs ]
    |       ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall`.

Near pkgs/development/node-packages/node-env.nix:417:7:
```
    |
417 |       installPhase = ''
    |       ^
```
</li>
  </ul>
</details>

File: pkgs/servers/matrix-synapse/matrix-appservice-irc/REVISION [hunk 3]
```diff
@@ -0,0 +1 @@
+0.30.0
```
Comment (SuperSandro2000) [inline pkgs/servers/matrix-synapse/matrix-appservice-irc/REVISION:1]: Nix can return this value from the derivation eg ``nix-instantiate --eval -E 'with (import ./. { }).pkgs; python3Packages.pyopenssl.version'``

**Comment by piegamesde** [comment]
I've opened an [upstream issue](https://github.com/matrix-org/matrix-python-sdk/issues/323) for the test. In the meantime, downgrading to a lower version of the library should work.

**Comment by andir** [comment]
> I've opened an [upstream issue](https://github.com/matrix-org/matrix-python-sdk/issues/323) for the test. In the meantime, downgrading to a lower version of the library should work.

Any reason we aren't just using `nio` instead? The python sdk even recommends that: https://github.com/matrix-org/matrix-python-sdk#project-status

**Comment by piegamesde** [comment]
This is purely historical (maybe due to cargo cult, maybe it *was* the best choice at the time); switching to a different library should be purely a matter of somebody taking the time and doing the work.

**Pull request closed without merging**